app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineRunErrorDetails.tsx

```diff
@@ -132,103 +191,112 @@ interface IErrorEntry {
   supportedDocumentationUrl?: string;
 }
 
-export default function PipelineRunErrorDetails() {
-  const [detailsExpanded, setDetailsExpanded] = useState<boolean>(false);
-  const [logsOpened, setLogsOpened] = useState<boolean>(false);
+interface IErrorDetailsBannerViewProps {
+  dataFetcher: IDataFetcher;
+  hasSurvey?: boolean;
+  errorDetails?: IErrorEntry[];
+  previewId?: string;
+}
 
-  const appId = useSelector((state) => state.name);
-  const artifactName = useSelector((state) => state.artifact.name);
-  const currentRun = useSelector((state) => state.currentRun);
-  const runid = currentRun?.runid;
-  const loading = useSelector((state) => state.runErrorDetailsLoading[runid] || false);
-  const errorDetails = useSelector((state) => state.runErrorDetails[runid]);
-  const canExapndErrorDetails = !!errorDetails?.length;
+interface IErrorDetailsBannerUiState {
+  detailsExpanded: boolean;
+  highlightedStage: string;
+}
 
-  const dispatch = useDispatch();
-  function setLoading(value: boolean) {
-    dispatch({
-      type: ACTIONS.SET_RUN_ERROR_DETAILS_LOADING,
-      payload: {
-        runid,
-        value,
-      },
-    });
-  }
+const errorDetailsBannerUiState: IErrorDetailsBannerUiState = {
+  detailsExpanded: false,
+  highlightedStage: '',
+};
 
-  function setErrorDetails(errors: IErrorEntry[]) {
-    dispatch({
-      type: ACTIONS.SET_RUN_ERROR_DETAILS,
-      payload: {
-        runid,
-        errors,
-      },
-    });
+const ACTION_PREFIX = 'ERROR_DETAILS_BANNER_ACTIONS';
+const ErrorDetailsBannerUiActions = {
+  EXPAND_ERROR_DETAILS: `${ACTION_PREFIX}/EXPAND_ERROR_DETAILS`,
+  COLLAPSE_ERROR_DETAILS: `${ACTION_PREFIX}/COLLAPSE_ERROR_DETAILS`,
+  SET_HIGHLIGHTED_STAGE: `${ACTION_PREFIX}/SET_HIGHLIGHTED_STAGE`,
+  RESET_HIGHLIGHTED_STAGE: `${ACTION_PREFIX}/RESET_HIGHLIGHTED_STAGE`,
+  RESET: `${ACTION_PREFIX}/RESET`,
+};
+
+function errorDetailsBannerUiReducer(
```

Comment:
Move to a separate file? It's going to be hard to follow the code if some reducers/stores are in their own files, but others are embedded with UI components.

Reply:
Done.